### ISSUE TITLE
Fuzzy Boolean Operations

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5612]
Issue imported from https://tracker.freecad.org/view.php?id=1967

* **Reporter:** shoogen
* **Date submitted:** 2/14/2015
* **FreeCAD version:** 
* **Category:** Feature
* **Status:** assigned
* **Tags:** 

# Original report text

make use of Fuzzy Boolean Operations
http://dev.opencascade.org/index.php?q=node/1056

# Other bug information

* **Priority:** none
* **Severity:** feature
* **Category:** Feature
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by shoogen 2015-03-22 22:21
git://github.com/5263/FreeCAD dev-fuzzyboolean

The development has stalled. Please feel free to test and comment. I would like to wait with the integration until the problem with corrupted input shapes (comment-520) is solved.

---

### Comment by openBrain 2019-08-31 10:57
As of today, fuzzy boolean operations are available in the Defeaturing WB.

**Comment by luzpaz** [comment]
@0penBrain what do you think, should we close this ticket ? I just wonder how people will learn of defeaturing wb just using vanilla freecad...you know?

**Comment by 0penBrain** [comment]
@luzpaz have no clue about the ticket. Just depends if we want this to be implemented in FC or not. :) From a quick reading of the article, to me it looks like it could have some added value. ;)

**Comment by maxwxyz** [comment]
The idea is to implement this into Part / PartDesign WB for boolean operations?

**Comment by FEA-eng** [comment]
@FreeCAD/cad-working-group I think this would be really useful. Can you add it to your list?

**Comment by maxwxyz** [comment]
is this still relevant, I thought @AIRCAP has implemented this in https://github.com/FreeCAD/FreeCAD/pull/17119?
@FlachyJoe maybe you know  it better

**Comment by FEA-eng** [comment]
> is this still relevant, I thought [@AIRCAP](https://github.com/AIRCAP) has implemented this in [#17119](https://github.com/FreeCAD/FreeCAD/pull/17119)?

Yes, some internally determined level of fuzziness was implemented but it’s often insufficient and there should be a way to set higher tolerance manually. That’s what can be done with Fuzzy booleans in the Defeaturing workbench and what is requested here. But it doesn’t have to be a separate set of (fuzzy) boolean operations. It could be just a property for the existing booleans.

**Comment by FlachyJoe** [comment]
`TopoShape::makeElementBoolean` already has a `tolerance` argument (default to -1 so the AIRCAP auto fuzzy value is applied).
It's quite easy to add a property to objects which use it and change the calls.

**Comment by pierreporte** [comment]
@FEA-eng What would be the priority? Nice to have, P3, P2?

**Comment by FEA-eng** [comment]
> [@FEA-eng](https://github.com/FEA-eng) What would be the priority? Nice to have, P3, P2?

I think that it could help with many cases (fuzzy booleans in the Defeaturing wb helped me several times), especially regarding export and use of FreeCAD models e.g. in FEM. So I would choose P2.

**Comment by FlachyJoe** [comment]
IMHO it is urgent to wait: #21713 and #21574 will make this easiest to implement.

**Comment by maxwxyz** [comment]
The mentioned PR was merged
